sql/api/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/TimeMode.scala

```diff
@@ -24,3 +27,19 @@ case object NoTime extends TimeMode
 case object ProcessingTime extends TimeMode
 
 case object EventTime extends TimeMode
+
+object TimeModes {
+  def apply(timeMode: String): TimeMode = {
+    timeMode.toLowerCase(Locale.ROOT) match {
+      case "none" | "notime" =>
+        NoTime
+      case "processingTime" =>
```

Comment:
This needs to be lower case.

Reply:
Thanks, good catch! Will update, same for the latter one